crates/db_schema/src/impls/community.rs

```diff
@@ -193,6 +193,19 @@ impl Community {
       .await?;
     Ok(())
   }
+
+  pub async fn get_random_local_community(pool: &mut DbPool<'_>) -> Result<Self, Error> {
+    let conn = &mut get_conn(pool).await?;
+    sql_function!(fn random() -> Text);
+    community::table
+      .filter(community::local)
```

Comment:
The issue doesnt mention that it should return only local communities, and thats not clear from the endpoint name either. Based on the issue it should take a type param (`GET /api/v3/community/random?type_=All`)

Reply:
K, added ListingType to the query.